### ISSUE TITLE
evmVersion mismatch error

### ISSUE THREAD

**Issue by cgewecke** [nomiclabs/buidler#271]
The docs advise Truffle 5 plugin users to set their solc config like this.
```javascript
module.exports = {
  solc: { version: "0.5.2" }
};
```
However it looks like buidler automatically sets the `evmVersion` to 'petersburg' triggering this error:
```shell
Invalid EVM version requested.
Error BDLR600: Compilation failed
```
because solc didn't start supporting it until [`0.5.5`](https://github.com/ethereum/solidity/releases/tag/v0.5.5)? Not sure. It might be possible semver check the compiler version when the evmVersion isn't specified and pick the best guess. 

A workaround is to use `0.5.5` or later.

**Comment by alcuadrado** [comment]
Thanks for reporting this!



> It might be possible semver check the compiler version when the evmVersion isn't specified and pick the best guess.

This is a great idea!

**Comment by alcuadrado** [comment]
Fixed by #277
